cura/CuraPackageManager.py

```diff
@@ -0,0 +1,357 @@
+            self._installPackage(installation_package_data)
+        self._to_install_package_dict.clear()
+        self._saveManagementData()
+
+    # Checks the given package is installed. If so, return a dictionary that contains the package's information.
+    def getInstalledPackageInfo(self, package_id: str) -> Optional[dict]:
+        if package_id in self._to_remove_package_set:
+            return None
+
+        if package_id in self._to_install_package_dict:
+            package_info = self._to_install_package_dict[package_id]["package_info"]
+            package_info["is_bundled"] = False
+            return package_info
+
+        if package_id in self._installed_package_dict:
+            package_info = self._installed_package_dict.get(package_id)
+            package_info["is_bundled"] = False
+            return package_info
+
+        for section, packages in self.getAllInstalledPackagesInfo().items():
+            for package in packages:
+                if package["package_id"] == package_id:
+                    package_info = package
+                    return package_info
+
+        return None
+
+    def getAllInstalledPackagesInfo(self) -> dict:
+        installed_package_id_set = set(self._installed_package_dict.keys()) | set(self._to_install_package_dict.keys())
+        installed_package_id_set = installed_package_id_set.difference(self._to_remove_package_set)
+
+        managed_package_id_set = set(installed_package_id_set) | self._to_remove_package_set
+
+        # map of <package_type> -> <package_id> -> <package_info>
+        installed_packages_dict = {}
+        for package_id in installed_package_id_set:
+            if package_id in self._to_install_package_dict:
+                package_info = self._to_install_package_dict[package_id]["package_info"]
+            else:
+                package_info = self._installed_package_dict[package_id]
+            package_info["is_bundled"] = False
+
+            package_type = package_info["package_type"]
+            if package_type not in installed_packages_dict:
+                installed_packages_dict[package_type] = []
+            installed_packages_dict[package_type].append( package_info )
+
+            # We also need to get information from the plugin registry such as if a plugin is active
+            package_info["is_active"] = self._plugin_registry.isActivePlugin(package_id)
+
+        # Also get all bundled plugins
+        all_metadata = self._plugin_registry.getAllMetaData()
+        for item in all_metadata:
+            plugin_package_info = self.__convertPluginMetadataToPackageMetadata(item)
+            # Only gather the bundled plugins here.
+            package_id = plugin_package_info["package_id"]
+            if package_id in managed_package_id_set:
+                continue
+
+            plugin_package_info["is_bundled"] = True if plugin_package_info["author"]["name"] == "Ultimaker B.V." else False
```

Comment:
Not all bundled plug-ins have the author "Ultimaker B.V." Look at the g-code reader for instance.

I think this check needs to be removed. We will want to merge plug-ins by other people in the future too.

Reply:
In this case we need a new mechanic for determining which plugins are bundled. Open to ideas.

Reply:
You could check if their `__init__.py` file is located in one of the resource directories? If it is, then it is not bundled. If it isn't, then it is bundled. Not entirely obvious in code, but it is correct.